### ISSUE TITLE
0 mAP on Pedestrian class

### ISSUE THREAD

**Issue by tdzdog** [happinesslz/EPNet#26]
Thanks for your excellent job. I simply modified the CLASSES from "Car" to "Pedestrian" in the 
 config file "LI_Fusion_with_attention_use_ce_loss.yaml ". But the validation results become 0(both mAP and recall). I am wondering why? Is there any constraint about the class?

**Comment by abdelrahman-osama** [comment]
I believe this is because the trained model you are using does not have this class. You may need to train your own model for this class.

**Comment by tdzdog** [comment]
No, I am not doing inference. I am trying to retrain the model for the 'Pedestrian' class but my trained model got 0 mAP.

**Comment by MC-CV** [comment]
My problem is the same as yours,and i don't know how to solve it...

**Comment by happinesslz** [comment]
@tdzdog @MC-CV @abdelrahman-osama   You may also need revise the CLS_MEAN_SIZE in [LI_Fusion_with_attention_use_ce_loss.yaml](https://github.com/happinesslz/EPNet/blob/0123c341243846aa3b412addcb9e2c07fd305237/tools/cfgs/LI_Fusion_with_attention_use_ce_loss.yaml#L19) as "[[1.76255119,0.66068622,0.84422524]]".  I get the PED results of " 3D: &71.54(easy)  &61.89(mod)  &55.65(hard)"

**Comment by MC-CV** [comment]
> @tdzdog @MC-CV @abdelrahman-osama You may also need revise the CLS_MEAN_SIZE in [LI_Fusion_with_attention_use_ce_loss.yaml](https://github.com/happinesslz/EPNet/blob/0123c341243846aa3b412addcb9e2c07fd305237/tools/cfgs/LI_Fusion_with_attention_use_ce_loss.yaml#L19) as "[[1.76255119,0.66068622,0.84422524]]". I get the PED results of " 3D: &71.54(easy) &61.89(mod) &55.65(hard)"

thks a lot，and how about the Cyclist's CLS_MEAN_SIZE,could you please publish that?Thank you very much!

**Comment by MC-CV** [comment]
> > @tdzdog @MC-CV @abdelrahman-osama You may also need revise the CLS_MEAN_SIZE in [LI_Fusion_with_attention_use_ce_loss.yaml](https://github.com/happinesslz/EPNet/blob/0123c341243846aa3b412addcb9e2c07fd305237/tools/cfgs/LI_Fusion_with_attention_use_ce_loss.yaml#L19) as "[[1.76255119,0.66068622,0.84422524]]". I get the PED results of " 3D: &71.54(easy) &61.89(mod) &55.65(hard)"
> 
> thks a lot，and how about the Cyclist's CLS_MEAN_SIZE,could you please publish that?Thank you very much!

I use [[1.73, 0.6, 1.76]] for cyclist finally,thks a lot for helping us！

**Comment by happinesslz** [comment]
@MC-CV  Yes, it's right. I also  use [[1.73698127,0.59706367,1.76282397]] for Cyclist.

**Comment by tdzdog** [comment]
> @tdzdog @MC-CV @abdelrahman-osama You may also need revise the CLS_MEAN_SIZE in [LI_Fusion_with_attention_use_ce_loss.yaml](https://github.com/happinesslz/EPNet/blob/0123c341243846aa3b412addcb9e2c07fd305237/tools/cfgs/LI_Fusion_with_attention_use_ce_loss.yaml#L19) as "[[1.76255119,0.66068622,0.84422524]]". I get the PED results of " 3D: &71.54(easy) &61.89(mod) &55.65(hard)"

Thank you very much！May I ask another question? I notice that you train models of different classes separately but the OpenPCDet (the newest official repo of PointRCNN, PV-RCNN and etc) has changed into multiple-class detection. Is there a performance difference between single-class and multiple-class models?

**Comment by MC-CV** [comment]
> > @tdzdog @MC-CV @abdelrahman-osama You may also need revise the CLS_MEAN_SIZE in [LI_Fusion_with_attention_use_ce_loss.yaml](https://github.com/happinesslz/EPNet/blob/0123c341243846aa3b412addcb9e2c07fd305237/tools/cfgs/LI_Fusion_with_attention_use_ce_loss.yaml#L19) as "[[1.76255119,0.66068622,0.84422524]]". I get the PED results of " 3D: &71.54(easy) &61.89(mod) &55.65(hard)"
> 
> Thank you very much！May I ask another question? I notice that you train models of different classes separately but the OpenPCDet (the newest official repo of PointRCNN, PV-RCNN and etc) has changed into multiple-class detection. Is there a performance difference between single-class and multiple-class models?

The two classification and multi classification loss functions are different, which will certainly have a certain impact on the results, but may have little impact. Recently, I plan to change the author's code to try to detect multi classification.
But I trained 50 epochs today ,and got low mAP. I don't know if it's the problem with other parameters. I just changed LR and batch size.

**Comment by tdzdog** [comment]
> > > @tdzdog @MC-CV @abdelrahman-osama You may also need revise the CLS_MEAN_SIZE in [LI_Fusion_with_attention_use_ce_loss.yaml](https://github.com/happinesslz/EPNet/blob/0123c341243846aa3b412addcb9e2c07fd305237/tools/cfgs/LI_Fusion_with_attention_use_ce_loss.yaml#L19) as "[[1.76255119,0.66068622,0.84422524]]". I get the PED results of " 3D: &71.54(easy) &61.89(mod) &55.65(hard)"
> > 
> > 
> > Thank you very much！May I ask another question? I notice that you train models of different classes separately but the OpenPCDet (the newest official repo of PointRCNN, PV-RCNN and etc) has changed into multiple-class detection. Is there a performance difference between single-class and multiple-class models?
> 
> The two classification and multi classification loss functions are different, which will certainly have a certain impact on the results, but may have little impact. Recently, I plan to change the author's code to try to detect multi classification.
> But I trained 50 epochs today ,and got low mAP. I don't know if it's the problem with other parameters. I just changed LR and batch size.

Thanks. I did not notice that the loss is different. I will have a try.

**Comment by MC-CV** [comment]
> @tdzdog @MC-CV @abdelrahman-osama You may also need revise the CLS_MEAN_SIZE in [LI_Fusion_with_attention_use_ce_loss.yaml](https://github.com/happinesslz/EPNet/blob/0123c341243846aa3b412addcb9e2c07fd305237/tools/cfgs/LI_Fusion_with_attention_use_ce_loss.yaml#L19) as "[[1.76255119,0.66068622,0.84422524]]". I get the PED results of " 3D: &71.54(easy) &61.89(mod) &55.65(hard)"

Are you sure that 3D PED have such high indicators? I see that the highest on Kitti is 46.88% (MOD) 52.75% (easy) 43.50% (hard), and after configuration, I run 50 epochs, and the results are only about 10%.

**Comment by happinesslz** [comment]
@MC-CV  It is noted that the  PED detection performance with 3D mAP of "&71.54(easy) &61.89(mod) &55.65(hard)" is obtained on the validation set, not test benchmark.
